src/solver/infeasible-problem-analysis/include/antares/solver/infeasible-problem-analysis/report.h

```diff
@@ -38,20 +42,18 @@ class InfeasibleProblemReport
 public:
     InfeasibleProblemReport() = delete;
     explicit InfeasibleProblemReport(
-      const std::vector<const operations_research::MPVariable*>& slackVariables);
-    void prettyPrint();
-
-private:
-    void turnSlackVarsIntoConstraints(
-      const std::vector<const operations_research::MPVariable*>& slackVariables);
-    void sortConstraintsBySlackValue();
-    void trimConstraints();
-    void sortConstraintsByType();
+      const std::vector<const operations_research::MPVariable*>& slackVariables,
+      const std::vector<std::shared_ptr<WatchedConstraint>>&);
     void logSuspiciousConstraints();
     void logInfeasibilityCauses();
 
-    std::vector<Constraint> constraints_;
-    std::map<ConstraintType, unsigned int> nbConstraintsByType_;
-    const unsigned int nbMaxVariables = 10;
+private:
+    void buildConstraintsFromSlackVars();
+    void filterConstraintsToOneByType();
+
+    const std::vector<std::shared_ptr<WatchedConstraint>>& constraintTypes_;
+    const std::vector<const MPVariable*>& slackVariables_;
+    std::vector<std::shared_ptr<WatchedConstraint>> constraints_;
```

Comment:
3 times std::vector<std::shared_ptr<WatchedConstraint>>
what's went wrong!?

Reply:
You right. This part is now simpler, some data members were removed.